extensions/amp-story/1.0/amp-story.js

```diff
@@ -713,6 +738,36 @@ export class AmpStory extends AMP.BaseElement {
       true /** callToInitialize */
     );
 
+    this.storeService_.subscribe(
+      StateProperty.SUBSCRIPTIONS_STATE,
+      (subscriptionsState) => {
+        this.subscriptionsStatePromise_.resolve();
+        if (subscriptionsState === SubscriptionsState.GRANTED) {
```

Comment:
maybe we should do this in the other extension? I don't feel strongly

Reply:
I actually think it can be better to put it here since we now only have one handler for the subscription state change, whereas we need to have two handlers for dialog ui state change separate in amp-story and amp-story-subscriptions. This also makes the story navigation logic in this file more complete, e.g. the way it works from blocked to granted status.

Reply:
ok sgtm